src/pb_memory.cc

```diff
@@ -141,15 +141,36 @@ PbMemory::CopyBuffer(
     kind = cudaMemcpyDeviceToDevice;
   }
 
-  cudaError_t err =
-      cudaMemcpy(dst->DataPtr(), src->DataPtr(), src->ByteSize(), kind);
+  cudaError_t err;
+  if ((src->MemoryType() == TRITONSERVER_MEMORY_GPU &&
```

Comment:
can we reuse the kind variable above and make this if kind ==cudaMemcpyDeviceToDevice

Reply:
+1 on this

Reply:
fixed.